src/main/kotlin/ink/abb/pogo/scraper/tasks/LootOneNearbyPokestop.kt

```diff
@@ -27,54 +30,57 @@ class LootOneNearbyPokestop(val sortedPokestops: List<Pokestop>, val lootTimeout
     override fun run(bot: Bot, ctx: Context, settings: Settings) {
         // STOP WALKING! until loot is done
         ctx.pauseWalking.set(true)
-        ctx.api.setLocation(ctx.lat.get(), ctx.lng.get(), getAltitude(ctx.lat.get(), ctx.lng.get(),ctx))
+        ctx.api.setLocation(ctx.lat.get(), ctx.lng.get(), getAltitude(ctx.lat.get(), ctx.lng.get(), ctx))
         val nearbyPokestops = sortedPokestops.filter {
-            it.canLoot(lootTimeouts = lootTimeouts, api = ctx.api)
+            it.canLoot(lootTimeouts = lootTimeouts)
         }
 
         if (nearbyPokestops.isNotEmpty()) {
             val closest = nearbyPokestops.first()
             var pokestopID = closest.id
-            if (settings.displayPokestopName)
-                pokestopID = "\"${closest.details.name}\""
-            Log.normal("Looting nearby pokestop $pokestopID")
-            val result = try {
-                closest.loot()
-            } catch (e: Exception) {
-                null
+            if (settings.displayPokestopName) {
+                if (!closest.fetchedDetails) {
+                    val countdownFetch = CountDownLatch(1)
+                    bot.api.queueRequest(closest.getFortDetails()).subscribe { countdownFetch.countDown() }
+                    countdownFetch.await()
+                }
+                pokestopID = "\"${closest.name}\""
             }
+            Log.normal("Looting nearby pokestop $pokestopID")
 
-            if (result == null) {
+            val result = closest.loot().toBlocking().first().response
+
+            if (result === null) {
                 // unlock walk block
                 ctx.pauseWalking.set(false)
                 return
             }
-            ctx.pokestops.getAndAdd(1)
-            if (result.itemsAwarded != null) {
-                ctx.itemStats.first.getAndAdd(result.itemsAwarded.size)
+
+            if (result.itemsAwardedCount != 0) {
+                ctx.itemStats.first.getAndAdd(result.itemsAwardedCount)
             }
 
-            if (result.experience > 0) {
+            if (result.experienceAwarded > 0) {
                 ctx.server.sendProfile()
             }
 
             when (result.result) {
                 Result.SUCCESS -> {
                     ctx.server.sendPokestop(closest)
                     ctx.server.sendProfile()
-                    var message = "Looted pokestop $pokestopID; +${result.experience} XP"
+                    var message = "Looted pokestop $pokestopID; +${result.experienceAwarded} XP"
                     if (settings.displayPokestopRewards)
-                        message += ": ${result.itemsAwarded.groupBy { it.itemId.name }.map { "${it.value.size}x${it.key}" }}"
+                        message += ": ${result.itemsAwardedList.groupBy { it.itemId.name }.map { "${it.value.size}x${it.key}" }}"
                     Log.green(message)
                     lootTimeouts.put(closest.id, closest.cooldownCompleteTimestampMs)
```

Comment:
Is it possible this doesn't work (anymore)? The bot tries to loot almost every pokestop twice (because it's the closest?) which results in a `Result.IN_COOLDOWN_PERIOD` and blacklist